streams/src/test/java/org/apache/kafka/streams/tests/SmokeTestDriver.java

```diff
@@ -252,93 +194,54 @@ private static void shuffle(final int[] data, final int windowSize) {
+        props.put(ConsumerConfig.KEY_DESERIALIZER_CLASS_CONFIG, StringDeserializer.class);
         props.put(ConsumerConfig.VALUE_DESERIALIZER_CLASS_CONFIG, ByteArrayDeserializer.class);
 
-        final KafkaConsumer<byte[], byte[]> consumer = new KafkaConsumer<>(props);
-        final List<TopicPartition> partitions = getAllPartitions(consumer, "echo", "max", "min", "dif", "sum", "cnt", "avg", "wcnt", "tagg");
+        final KafkaConsumer<String, byte[]> consumer = new KafkaConsumer<>(props);
+        final String[] topics = {"data", "echo", "max", "min", "dif", "sum", "cnt", "avg", "tagg"};
+        final List<TopicPartition> partitions = getAllPartitions(consumer, topics);
         consumer.assign(partitions);
         consumer.seekToBeginning(partitions);
 
-        final int recordsGenerated = allData.size() * maxRecordsPerKey;
+        final int recordsGenerated = inputs.size() * maxRecordsPerKey;
         int recordsProcessed = 0;
+        final Map<String, AtomicInteger> processed =
+            Stream.of(topics)
+                  .collect(Collectors.toMap(t -> t, t -> new AtomicInteger(0)));
+
+        final Map<String, Map<String, LinkedList<Number>>> events = new HashMap<>();
 
-        final HashMap<String, Integer> max = new HashMap<>();
-        final HashMap<String, Integer> min = new HashMap<>();
-        final HashMap<String, Integer> dif = new HashMap<>();
-        final HashMap<String, Long> sum = new HashMap<>();
-        final HashMap<String, Long> cnt = new HashMap<>();
-        final HashMap<String, Double> avg = new HashMap<>();
-        final HashMap<String, Long> wcnt = new HashMap<>();
-        final HashMap<String, Long> tagg = new HashMap<>();
-
-        final HashSet<String> keys = new HashSet<>();
-        final HashMap<String, Set<Integer>> received = new HashMap<>();
-        for (final String key : allData.keySet()) {
-            keys.add(key);
-            received.put(key, new HashSet<Integer>());
-        }
-        int retry = 0;
         final long start = System.currentTimeMillis();
         while (System.currentTimeMillis() - start < TimeUnit.MINUTES.toMillis(6)) {
-            final ConsumerRecords<byte[], byte[]> records = consumer.poll(Duration.ofMillis(500));
+            final ConsumerRecords<String, byte[]> records = consumer.poll(Duration.ofSeconds(30));
             if (records.isEmpty() && recordsProcessed >= recordsGenerated) {
-                if (verifyMin(min, allData, false)
-                    && verifyMax(max, allData, false)
-                    && verifyDif(dif, allData, false)
-                    && verifySum(sum, allData, false)
-                    && verifyCnt(cnt, allData, false)
-                    && verifyAvg(avg, allData, false)
-                    && verifyTAgg(tagg, allData, false)) {
-                    break;
-                }
-                if (retry++ > MAX_RECORD_EMPTY_RETRIES) {
-                    break;
-                }
+                break;
```

Comment:
The prior logic would keep polling for up to 30 seconds after we pulled as many records as we generated, as long as the verification kept failing.

If we pull more than we generated, it's already a failure, so we might as well fail fast. If you agree, then the only remaining logic was to poll for 30 seconds in .5 second increments. Now that we have long polling in the consumer, we might as well just poll for up to 30 seconds once.

Reply:
Just one minor point, if we don't have EOS enabled, we could end up pulling duplicates in some tests like the `StreamsUpgradeTest::upgrade_downgrade_brokers`

Reply:
This is a good point.

I have noticed in running the test locally that the long-poll does seem slower than our prior code here, so maybe I'll go ahead and revert this part of the diff.